src/modules/wallet/utils.ts

```diff
@@ -201,11 +201,23 @@ export const createTransaction = async (body: CreateTransactionParams) => {
 /**
  * @returns the first unused address for a wallet.
  */
-export const getUnusedAddress = async (walletName: string): Promise<string> => {
-  const addresses = await window.Wallets[walletName].getUnusedAddresses();
-  const encodedAddress = addressFromHex(addresses[0]);
+export const getWalletAddress = async (walletName: string): Promise<string> => {
+  let addresses = [];
+  const wallet = window.Wallets[walletName];
+
+  // prefer unused address, but some wallets such as Nami require used address
+  addresses = await wallet.getUnusedAddresses();
+  if (addresses.length === 0) {
+    addresses = await wallet.getUsedAddresses();
```

Comment:
Up to you, but this could be simplified to just always call `getUsedAddresses()` regardless of whether its Nami or not. When we end up sending tokens back to them, we will always be sending to a used address anyway.

Sending to the very first address in the wallet prevents a specific type of attack called a franken-address attack. Our current sale doesn't make sense to attack this way, but if we ever implement something like whitelisting, it prevents someone from hacking themselves onto the whitelist pretending to be someone else.